### ISSUE TITLE
Add a /health endpoint which does not require authentication

### ISSUE THREAD

**Issue by anguslees** [restic/rest-server#147]
Trying to deploy restic/rest-server:0.10.0 in k8s.  I want a URL endpoint that I can use for "livenessProbe" (GET, should return 200).

As far as I can see from some testing, guided by mux.go - everything returns 401 Unauthorized.  At the moment, I'm using a naive tcp 'liveness' probe, which is not nearly as expressive/accurate.

It would be nice if there was a /health endpoint (or similar) which did not require authentication, and returned 200 (assuming the server was ok).

**Comment by Enrico204** [comment]
I created a WIP pull request with a proposal for a `/health` endpoint. For now, the endpoint checks for free space (at least 8MB *) and that the repository path is writable. Should we check for more things?

Maybe we can add some other checks when/if external auth backends will be implemented (see #111 and #70 )

* I picked the value as the size on one pack, however I don't know if it's correct. Feel free to propose a different value

**Comment by MichaelEischer** [comment]
Which checks do we really need for a `/health` endpoint? Checking whether there is enough free space to store at least one pack file is mostly of academic interest. After all it will only take a very short time to fill up the free space completely at which point the server will stop accepting new uploads. And I don't think that there's a right answer to what the limit should be, probably every admin will want to use different limits.

If I understand liveness probes correctly, they are used to restart stuck containers. That is a failed health probe would cause container restarts. However, restarting the rest-server container once a disk has run full is highly problematic as that would prevent (read) access to the backup repositories.

And I guess a similar reasoning applies to whether the repository path is writeable (although that might be less of a problem).

**Comment by anguslees** [comment]
Agreed, the focus should be on whether killing+restarting this container would help. This isn't a substitute for more comprehensive monitoring.

A quite reasonable first step is to do no extra logic and just respond with 200 ok immediately, from your main http event handler.  Even that trivial check still confirms that the program is running, is listening on the correct port, has completed any startup steps, isn't in deadlock or oom-thrash, etc.

Just for completeness, _don't_ make 'healthiness' depend on reachability/health of some _other_ remote service.  This is a common error and leads to cascading failures.

**Comment by queeup** [comment]
I would love to have this for checking if my rest-server up before triggering remote backup with systemd.

**Comment by rudolfbyker** [comment]
When deploying this on Kubernetes, there are three probes:
- `readinessProbe`: Check whether the container is ready to accept HTTP requests. A simple `/ping` endpoint that checks nothing and always responds with 200 OK is sufficient for this, since we only want to check if the HTTP server is up.
- `livenessProbe` (mentioned above): Check whether the container should be restarted.
- `startupProbe`: Check whether the container is still starting up.

The `readinessProbe` is the easiest. Let's start there, and create new issues for discussion `livenessProbe` and `startupProbe`, since they are conceptually more complicated to get right.

See https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
